apollo/lib/schemas/user.py

```diff
@@ -43,13 +48,24 @@ class UserSchema(ORMBase):
 
 
 class UpdateUserSchema(BaseModel):
-    old_password: constr(min_length=8, strip_whitespace=True)
-    password: constr(min_length=8, strip_whitespace=True)
-    password_confirm: constr(min_length=8, strip_whitespace=True)
+    username: Optional[
+        constr(min_length=1, max_length=36, strip_whitespace=True)]
+    password: Optional[constr(min_length=8, strip_whitespace=True)]
+    password_confirm: Optional[constr(min_length=8, strip_whitespace=True)]
+    old_password: Optional[constr(min_length=8, strip_whitespace=True)]
 
-    @validator('password')
+    class Config:
+        arbitrary_types_allowed = True
+
+    @validator('username')
     @classmethod
-    def no_whitespace_in_password(cls, value):
+    @with_db_session
+    def unique_username(cls, value, **kwargs):
+        return check_for_unique_username(value, kwargs['session'])
+
+    @validator('password', 'username')
+    @classmethod
+    def no_whitespace(cls, value, values):
```

Comment:
```suggestion
    def no_whitespace(cls, value):
```

For future reference: I don't think you need to take `values` if you don't need it.